android/app/src/main/java/net/pengcook/android/presentation/core/util/ImageUtils.kt

```diff
@@ -2,68 +2,126 @@ package net.pengcook.android.presentation.core.util
         )
     }
 
     fun createImageFile(): File {
         val timeStamp: String =
-            SimpleDateFormat("yyyyMMdd_HHmmss", Locale.getDefault()).format(Date())
+            SimpleDateFormat(DATA_FORMAT, Locale.getDefault()).format(Date())
         val storageDir: File? = context.getExternalFilesDir(null)
         return File
             .createTempFile(
                 "JPEG_${timeStamp}_",
-                ".jpg",
+                FILE_SUFFIX,
                 storageDir,
             )
     }
 
-    fun getUriForFile(file: File): Uri {
-        return FileProvider.getUriForFile(
+    fun getUriForFile(file: File): Uri =
+        FileProvider.getUriForFile(
             context,
             "net.pengcook.android.fileprovider",
             file,
         )
-    }
 
-    fun processImageUri(uri: Uri): String? {
-        return try {
+//    fun processImageUri(uri: Uri): String? =
+//        try {
+//            val inputStream = context.contentResolver.openInputStream(uri)
+//            if (inputStream != null) {
+//                val tempFile = createTempImageFile()
+//                tempFile.outputStream().use { outputStream ->
+//                    inputStream.copyTo(outputStream)
+//                }
+//                tempFile.absolutePath
+//            } else {
+//                null
+//            }
+//        } catch (e: IOException) {
+//            e.printStackTrace()
+//            null
+//        }
+
+    fun isPermissionGranted(permissions: Array<String>): Boolean =
+        permissions.all {
+            ContextCompat.checkSelfPermission(context, it) == PackageManager.PERMISSION_GRANTED
+        }
+
+    suspend fun compressAndResizeImage(uri: Uri): File =
+        withContext(Dispatchers.IO) {
             val inputStream = context.contentResolver.openInputStream(uri)
-            if (inputStream != null) {
-                val tempFile = createTempImageFile()
-                tempFile.outputStream().use { outputStream ->
-                    inputStream.copyTo(outputStream)
-                }
-                tempFile.absolutePath
-            } else {
-                null
-            }
-        } catch (e: IOException) {
-            e.printStackTrace()
-            null
+            val originalBitmap = BitmapFactory.decodeStream(inputStream)
+
+            val resizedBitmap =
+                adjustImageOrientation(
+                    Bitmap.createScaledBitmap(
+                        originalBitmap,
+                        MAX_WIDTH,
+                        MAX_HEIGHT,
+                        true,
+                    ),
+                    uri,
+                )
+
+            val compressedFile = createTempImageFile()
+            val outputStream = FileOutputStream(compressedFile)
+            resizedBitmap.compress(Bitmap.CompressFormat.JPEG, COMPRESSED_QUALITY, outputStream)
+            outputStream.flush()
+            outputStream.close()
+
+            originalBitmap.recycle()
```

Comment:
> This is an advanced call, and normally need not be called, since the normal GC process will free up this memory when there are no more references to this bitmap.

인터페이스 문서 상에서, GC가 알아서 처리해줘서 recycle()은 딱히 호출할 필요는 없다고 하는데 지우는 것은 어떻게 생각하시나요?

Reply:
혹시라도 메모리 누수가 있을 것 같아서 바로 없애려고 한건데 우선 지워보겠습니다!